### ISSUE TITLE
交叉熵损失函数如何加权

### ISSUE THREAD

**Issue by dkerjw1** [PaddlePaddle/Paddle#19158]
- 版本、环境信息：
    1）PaddlePaddle版本：fluid
    2）CPU/GPU：GPU Cuda8.0
    3）系统环境：CentOS
    4）Python2.7

你好，请问如何实现在多分类时，使用加权的交叉熵函数，权重为类别的比例

**Comment by SunGaofeng** [comment]
可以考虑在fluid.layers.cross_entropy里面使用soft_label，将label乘以权重做，具体使用可以参考这个api的文档
https://www.paddlepaddle.org.cn/documentation/docs/zh/1.5/api_cn/layers_cn/nn_cn.html#cross-entropy

**Comment by dkerjw1** [comment]
请问有相关示例么

**Comment by SunGaofeng** [comment]
目前的cross_entropy op还不支持加权

**Comment by dkerjw1** [comment]
> 目前的cross_entropy op还不支持加权

那请问可以自定义loss么

**Comment by SunGaofeng** [comment]
可以，比如用下面这种方法定义带weight的loss
"""
/////for example 
/////        input  [N, D],  
/////        labels [N, D], which is one-hot along D 
/////        weights [N, D]
"""

log_input = fluid.layers.log(input)
log_input = 0. - labels * log_input
loss = log_input * weights
loss = fluid.layers.reduce_sum(loss, dim=1)
